### ISSUE TITLE
String variable values never escaped before saving to SQL

### ISSUE THREAD

**Issue by HeroGarr** [HerculesWS/Hercules#1373]
What the title says. If you try to save a string variable with a single quote inside , for instance, you'll be getting errors in char console.

Even if you screen them yourself during setting, script engine will unescape it, and you'll still be getting errors.

**Comment by Emistry** [comment]
you mean NPC script ? there already exist the script command `escape_sql( <string> )` to escape the contents.

**Comment by HeroGarr** [comment]
Yes, that'd avoid the error for saving, but the problem is, if you'll use escape_sql to assign a line to variable until you relog it'll have screens displayed.

For instance, I use the variable to save last warp location name. In this case, **_Nydhoggur's dungeon_**.
![pic1](https://cloud.githubusercontent.com/assets/10124529/17135102/aaeb7d48-5336-11e6-986a-91ea8cc71464.jpg)

So, until I relog, screen is not taken away.
![pic2](https://cloud.githubusercontent.com/assets/10124529/17135144/deb4990c-5336-11e6-9519-e7a082c9000c.jpg)

**Comment by Emistry** [comment]
I believe it's just your NPC script issue. `escape_sql()` are used during the moment you try to save the variable value into the database.

You don't need to use it when you just want the value to be display in the npc dialog.

**Comment by HeroGarr** [comment]
Um... Actually, here's where you're wrong. escape_sql returns escaped string, it doesn't mark variable for "saving" in any kind.

And we actually don't have *ANY* control over saving variables. Moment it gets new assignment it's saved. So I either assign it with escaped string via escape_sql and have screen until relog, or I get an error and value is not saved at all. I don't think that's intended.

Test script:
`prontera,150,150,0	script	asdalasd	HIDDEN_NPC,{
mes "Current var is: " + var$;
var$ = "This string has ' symbol.";
mes "Pay attention to your char server log, it should provide an error.";
next;
var$ = escape_sql("This string has ' symbol.");
mes var$+"";
mes "Now variable shouldn't error on saving, but the above will contain \\.";
mes "Please relog, and you'll see in first message that variable doesn't have \\ anymore.";
close;
}`

Results:
![pic1](https://cloud.githubusercontent.com/assets/10124529/17563283/4c96970c-5f37-11e6-8a29-581dd23bd3b8.jpg)
![pic1 2](https://cloud.githubusercontent.com/assets/10124529/17563286/4e6a008c-5f37-11e6-903c-73e218dc6869.jpg)
![pic2](https://cloud.githubusercontent.com/assets/10124529/17563287/4ff7bfb6-5f37-11e6-913f-dfee73a6d28b.jpg)
![pic3](https://cloud.githubusercontent.com/assets/10124529/17563288/515eddc6-5f37-11e6-8559-d5e026b2f4b2.jpg)

**Comment by Emistry** [comment]
I see...

when I test with this NPC... **it seem like work fine**...
```
prontera,150,150,0	script	asdalasd	757,{
mes "Current var is: " + var$;
var$ = "This string has ' symbol.";
mes "Pay attention to your char server log, it should provide an error.";
next;
var$ = escape_sql("This string has ' symbol.");
mes var$+"";
mes "Now variable shouldn't error on saving, but the above will contain \\.";
mes "Please relog, and you'll see in first message that variable doesn't have \\ anymore.";
close;
}
```
however, **it should detect the error here** during the first time assign, but in my server it doesn't and it proceed the script without any errors.


if I test using this NPC .... **error shown**.
```
prontera,150,150,0	script	asdalasd	757,{
mes "Current var is: " + var$;
var$ = "This string has ' symbol.";
mes "Pay attention to your char server log, it should provide an error.";
close;
}
```

**Comment by MishimaHaruna** [comment]
I'm very surprised to see this (how come it was never noticed before?) but it doesn't look like we do any form of escaping (nor we use prepared statements, which would be advisable) in `inter_savereg()` and `mmo_save_accreg2()`. Ouch.

**Comment by MishimaHaruna** [comment]
This should be fixed via #1400
